instrumentation/net/http/otelhttpconv/DESIGN.md

```diff
@@ -0,0 +1,139 @@
+# OTel HTTP Conv
+
+## Motivation
+
+We provide many instrumentation libraries for HTTP packages and frameworks.
+Many of them need to reimplement similar features, as they have their own
+internals (they don't all allow using `net/http.Handler`).
+This is causing a lot of duplication across instrumentations, and makes
+standardization harder.
+
+Folks have also expressed interest in being able to use the internal tools
+provided by the internal `semconvutil` package within their own
+instrumentations ([#4580](https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4580)).
+This package aims to solve that by being publicly usable.
+
+This document outlines a proposal for a new instrumentation package called
+`otelhttpconv` that will provide a way to reduce some of this code repetition,
+especially with the setup of spans and metrics and their attributes.
```

Comment:
Do we want to depend on this module in other modules such as `otelhttp`, `otelmux` etc

If so then can you explicitly mention it in this document as well as add something about backwards compatibility? 

The reason why I did not want to come up with such module was that any breaking change of such module can break compilation of modules depending on this module.

Basically I am afraid of having problems like:
- https://github.com/open-telemetry/opentelemetry-go/issues/3846

Reply:
I've added a section about that.